File: NotesServer/Models/Connection.cs
using SQLite;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using System;

namespace NotesServer.Models
{
    public class Connection : DatabaseEntity
    {
        [PrimaryKey, AutoIncrement]
        [JsonPropertyName("id")]
        public int Id { get; set; }

        [JsonPropertyName("fromNoteId")]
        public int FromNoteId { get; set; }

        [JsonPropertyName("toNoteId")]
        public int ToNoteId { get; set; }

        [Required]
        [JsonPropertyName("type")]
        public string Type { get; set; } = string.Empty;

        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;

        [JsonPropertyName("connectionColor")]
        public int ConnectionColor { get; set; }

        [Required]
        public string ConnectionString { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
    }
} 

File: NotesServerGO/models/connection.go
package models

import "time"

// Connection представляет связь между двумя заметками.
type Connection struct {
	Id              int64     `json:"id" db:"Id"`
	FromNoteId      int64     `json:"from_note_id" db:"FromNoteId"` // ID заметки-источника
	ToNoteId        int64     `json:"to_note_id" db:"ToNoteId"`     // ID заметки-цели
	Name            string    `json:"name" db:"Name"`
	ConnectionColor int       `json:"connection_color" db:"ConnectionColor"` // ARGB int
	DatabaseId      int64     `json:"database_id" db:"DatabaseId"`
	CreatedAt       time.Time `json:"-" db:"CreatedAt"`
	UpdatedAt       time.Time `json:"-" db:"UpdatedAt"`
}
